File: Unwritten/4C.py
database = {}

ask = int(input())

for i in range(0, ask):
    test = str(input())
    database.setdefault(test, -1)
    database[test] += 1

    if database[test] == 0:
        print("OK")
    else:
        print(test + str(database[test]))

File: Unwritten/4C.cpp
#include <bits/stdc++.h>
using namespace std;

int main(){
    int n;
    cin >> n;
    map<string,int> database;

    while(n--){
        string test;
        cin >> test;
        if (database.count(test) == 0){
            cout << "OK"<<endl;
            database[test] = 1;

        }else{
            cout << test << database[test] << endl;
            database[test] += 1;
        }
    }
}
